library/agent/context/user.ts

```diff
@@ -49,10 +49,8 @@ export function setUser(user: unknown) {
   }
 
   if (context.executedMiddleware) {
-    // eslint-disable-next-line no-console
-    console.warn(
-      `setUser(...) must be called before the Zen middleware is executed.`
-    );
+    logWarningSetUserCalledAfterMiddleware();
+    return;
```

Comment:
Do we want to return here? Setting the user will still affect the users sent to core.